lib/api/salesforceChatAPI.js

```diff
@@ -0,0 +1,47 @@
+import { NativeModules } from "react-native";
+
+export default class SalesforceChatAPI {
+  constructor() {
+    this.salesforceChat = NativeModules.RNSalesforceChat;
+  }
+
+  async createUserData(chatUserData) {
+    this.salesforceChat.createPreChatData(
+      chatUserData.agentLabel,
+      chatUserData.value,
+      chatUserData.isDisplayedToAgent
+    );
+  }
```

Comment:
Since we have no control about the user's input, I think we should have a clear validation in the library setup.

```javascript
async createUserData(chatUserData) {
   const { agentLabel, value, isDisplayedToAgent } = chatUserData

  if (!agentLabel) {
    throw new Error('required field agentLabel')
  }

  if (!value) {
    throw new Error('required field value')
  }

  if (!isDisplayedToAgent) {
    throw new Error('required field isDisplayedToAgent')
  }

    this.salesforceChat.createPreChatData(
      chatUserData.agentLabel,
      chatUserData.value,
      chatUserData.isDisplayedToAgent
    );
  }
```

This way, if the user types something wrong or forget to pass any argument, we show a clear error and exempt us from the responsibility of having to fix the library.

Reply:
@pedrohperalta Great idea!